File: Source/Java/Expressive/src/main/java/com/bijington/expressive/operators/grouping/ParenthesisCloseOperator.java
package com.bijington.expressive.operators.grouping;

import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.operators.OperatorBase;
import com.bijington.expressive.operators.OperatorPrecedence;

/**
 * Created by shaun on 10/07/2016.
 */
public class ParenthesisCloseOperator extends OperatorBase {
    @Override
    public String[] getTags() {return new String[] { ")" }; }

    @Override
    public IExpression buildExpression(String previousToken, IExpression[] expressions) {
        return (expressions[0] != null) ? expressions[0] : expressions[1];
    }

    @Override
    public int getPrecedence(String previousToken) {
        return OperatorPrecedence.ParenthesisClose;
    }
}

File: Source/Expressive/Operators/Grouping/ParenthesisCloseOperator.cs
using Expressive.Expressions;

namespace Expressive.Operators.Grouping
{
    internal class ParenthesisCloseOperator : OperatorBase
    {
        #region OperatorBase Members

        public override string[] Tags => new[] { ")" };

        public override IExpression BuildExpression(Token previousToken, IExpression[] expressions, Context context)
        {
            return expressions[0] ?? expressions[1];
        }

        public override OperatorPrecedence GetPrecedence(Token previousToken)
        {
            return OperatorPrecedence.ParenthesisClose;
        }

        #endregion
    }
}
